### ISSUE TITLE
arg_max  error

### ISSUE THREAD

**Issue by linyia01** [tensorflow/tensorflow#11841]
my code is

y=tf.Variable(tf.random_normal(shape=[2,5,3,10,12 ,5 ]))
a=tf.arg_max(y,0)
a.shape
sess=tf.Session()
sess.run(tf.global_variables_initializer())
sess.run(a)

error is :
InvalidArgumentError (see above for traceback): ArgOp : Unhandled input dimensions: 6
	 [[Node: ArgMax_7 = ArgMax[T=DT_FLOAT, Tidx=DT_INT32, _device="/job:localhost/replica:0/task:0/cpu:0"](Variable_9/read, ArgMax_7/dimension)]]

**Comment by reedwm** [comment]
`tf.arg_max` only handles tensors with a 5 or less dimensions. See [argmax_op.cc](https://github.com/tensorflow/tensorflow/blob/r1.2/tensorflow/core/kernels/argmax_op.cc#L91).

If you want to run `tf.arg_max` on higher dimension tensors, explain why and reopen this bug.

**Comment by danijar** [comment]
I have a tensor with dimensions `(batch, time, width, height, channels, logits)` to predict video with a softmax over possible pixel values. This is a typical use case in video prediction. This logit tensor goes into `tfp.distributions.Categorical`. However, calling mode on this distribution fails because it uses `tf.argmax()`.

**Comment by danijar** [comment]
Thanks!

**Comment by idofr** [comment]
Still seeing the same issue with tf.1.15
https://github.com/tensorflow/tensorflow/blob/r1.15/tensorflow/core/kernels/argmax_op.cc#L91

**Comment by reedwm** [comment]
TF 1.15 doesn't have the fix unfortunately. TF 2.1 is the first release with the fix.
